corehq/apps/data_interfaces/tasks.py

```diff
@@ -53,6 +54,7 @@ def get_ids_from_url(query_string, domain, couch_user):
         _request.user = couch_user.get_django_user()
         _request.domain = domain
         _request.couch_user.current_domain = domain
+        _request.can_access_all_locations = can_access_all_locations
```

Comment:
You can also use `couch_user.has_permission(domain, 'access_all_locations')` if you wanted to avoid passing in another parameter.